lint.go

```diff
@@ -32,11 +32,15 @@ const styleGuideBase = "https://golang.org/wiki/CodeReviewComments"
 // Note that it doesn't follow grammar rules but common sense instead.
 const symbolsCanEndASentence = ".?!;\"'`\\/|)}]>/#$"
 
-// hasValidEndOfSentence checks is s has a valid end of sentence.
+// hasValidEndOfSentence checks if s has a valid end of sentence.
 func hasValidEndOfSentence(s string) bool {
 	s = strings.TrimSpace(s)
 	length := len(s)
-	if (length > 0) {
```

Comment:
My college Bill sent me comments by email:

```
+func hasValidEndOfSentence(s string) bool {
+	s = strings.TrimSpace(s)
+	length := len(s)

rather than stack allocating to store len(s) take advantage of it being O(1)

+	if (length == 0) {

spurious parens

+		// Don't enforce valid end of sentence on empty comment.
+		// It must be enforced by other restrictions if any.
+		return true

as this returns, remove the else case and unindent.

+	} else {
+		last := s[length - 1]
+		for i, _ := range symbolsCanEndASentence {
+			if last == symbolsCanEndASentence[i] {

no reason to duplicate the variable symbolsCanEndASentence

+				return true
+			}
+		}
+	}
+	return false
+}

so this could be simplified to:

+func hasValidEndOfSentence(s string) bool {
+	s = strings.TrimSpace(s)
+	if len(s) == 0 {
+		// Don't enforce valid end of sentence on empty comment.
+		return true
+	}
+	last := s[len(s) - 1]
+	for _, c := range symbolsCanEndASentence {
+		if last == c {
+			return true
+		}
+	}
+	return false
+}
```

Reply:
Agree with most, at least my rule is if someone has comment, it needs to be improved somehow likely.
1) "Avoid length", well after all I agree, if this is not best practice in go. Apart from this for sure can be handle by optimizer (if it does not do it now, we must implement it).
2) "no reason to duplicate the variable symbolsCanEndASentence" In fact, I did it intentionally, range returns in this case index and `rune' and compare w/o cast returns error:
"invalid operation: last == c (mismatched types byte and rune)"
In this case, it's good to know what is the rule for compare byte and rune. I see that it's safe to cast 'c' to byte.
So, the code is really:
	last := s[len(s) - 1]
	for _, runeValue := range symbolsCanEndASentence {
		if last == byte(runeValue) {